react.md

```diff
@@ -0,0 +1,242 @@
+# Trunk Club React Styleguide
+
+This guide assumes that you are writing in the coffeescript-flavor of [JSX](https://github.com/jsdf/coffee-react-transform). It also assumes that you have a cursory understanding of what JSX is and why React uses it. See the [docs](http://facebook.github.io/react/docs/jsx-in-depth.html) for more details.
+
+## Structure of React Components
+You should structure your React Components like this to ensure consistency. Whether you're using ES6 or Coffeescript or PoopyScript, it should be ordered like this
+
+1. Attributes
+  1. If you are using a mixin that requires you to specify an attribute on the Component's prototype (and not on the instance itself), do that here as well
+
+  ```Coffeescript
+  ComponentName = React.createClass
+    displayName: 'ComponentName'
+    mixins: []
+    statics: {}
+    propTypes: {}
+    listen: {} # Custom attribute needed by a mixin
+  ```
+
+2. React lifecycle hooks
+
+  ```Coffeescript
+  getInitialState: -> {}
+  componentWillMount: ->
+  componentDidMount: ->
+  componentWillReceiveProps: (nextProps) ->
+  shouldComponentUpdate: (nextProps, nextState) -> true
+  componentWillUpdate: (nextProps, nextState) ->
+  componentDidUpdate: (prevProps, prevState) ->
+  ```
+
+3. Event handlers
+
+  ```Coffeescript
+  onClickBtn: (evt) -> # on(EventType)(SourceItem)
+                       # the React docs will use `handle(EventType)`.
+                       # Same thinking, different naming
```

Comment:
Just because that's what we've got in the other Backbone apps. I'm not attached to it, though. Using `handle` will make it easier for us to point to the React docs as an example (without needing to think as much about how it's different for us in our apps).